src/components/Checkbox.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+
+const noop = () => {};
+
+/**
+ * @typedef CheckboxProps
+ * @prop {boolean} [defaultChecked] - Should the checkbox be selected/checked
+ *  by default?
+ * @prop {string} name - The `name` of the checkbox. Will be used as `id` if
+ *   `id` not provided
+ * @prop {string} [id] - HTML ID for the checkbox element
+ * @prop {string} label - Checkbox label
+ * @prop {(checked: boolean) => any} [onChanged] - Callback when checkbox is
+ *   checked/unchecked
+ */
+
+/**
+ * A checkbox input
+ *
+ * @param {CheckboxProps} props
+ */
+export function Checkbox({
+  defaultChecked = false,
+  id,
+  name,
+  label,
+  onChanged,
+}) {
+  let [isChecked, setChecked] = useState(defaultChecked);
```

Comment:
Note: the state modeling in this component is based on the client's `Menu` component (roughly)...just trying to reuse an established pattern.